server/graphql/preference/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+use async_graphql::*;
+use graphql_core::{standard_graphql_error::validate_auth, ContextExt};
+use graphql_types::types::PreferencesNode;
+use service::auth::{Resource, ResourceAccessRequest};
+
+#[derive(Default, Clone)]
+pub struct PreferenceQueries;
+#[Object]
+impl PreferenceQueries {
+    pub async fn preferences(
+        &self,
+        ctx: &Context<'_>,
+        store_id: String,
+    ) -> Result<PreferencesNode> {
+        let user = validate_auth(
+            ctx,
+            &ResourceAccessRequest {
+                resource: Resource::QueryStorePreferences,
+                store_id: None,
+            },
+        )?;
+
+        let service_provider = ctx.service_provider();
+        let service_ctx = service_provider.context(store_id.to_string(), user.user_id)?;
+        let service = &service_provider.preference_service;
+
+        let prefs = service.get_preferences(&service_ctx, &store_id)?;
+
+        Ok(PreferencesNode::from_domain(prefs))
+    }
+}
+
+// TODO: separate query for central edit UI - all prefs, global and for each store, rather than the consolidated list
+
+// #[derive(Default, Clone)]
+// pub struct CentralPreferenceQueries;
+// #[Object]
+// impl CentralPreferenceQueries {
+//     pub async fn preferences(
+//         &self,
+//         ctx: &Context<'_>,
+//         store_id: String,
+//     ) -> Result<PreferencesResponse> {
+//         validate_auth(
+//             ctx,
+//             &ResourceAccessRequest {
+//                 resource: Resource::QueryVaccineCourse,
```

Comment:
Globals prefs should just be restrict to central server admin users me thinks 🤔

Reply:
yah coming in next pr :)